packages/common/src/feedback/edit.ts

```diff
@@ -0,0 +1,76 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import { IUserItemOptions, removeItem } from "@esri/arcgis-rest-portal";
+import { PropertyMapper } from "../core/_internal/PropertyMapper";
+import { setDiscussableKeyword } from "../discussions";
+import { getUniqueSlug } from "../items";
+import { getModel, updateModel } from "../models";
+import { IModel } from "../types";
+import { setEntityStatusKeyword } from "../utils/internal/setEntityStatusKeyword";
+import { computeProps } from "./_internal/computeProps";
+import { getPropertyMap } from "./_internal/getPropertyMap";
+import { IHubFeedback } from "../core/types/IHubFeedback";
+import { setDisplayMapKeyword } from "./utils.ts";
+
+/**
+ * @private
+ * Update a Hub Feedback obejct
+ * @param feedback the feedback to update
+ * @param requestOptions user request options
+ * @returns promise that resolves a IHubFeedback
+ */
+export async function updateFeedback(
+  feedback: IHubFeedback,
+  requestOptions: IUserRequestOptions
+): Promise<IHubFeedback> {
+  // verify that the slug is unique, excluding the current feedback
+  feedback.slug = await getUniqueSlug(
+    { slug: feedback.slug, existingId: feedback.id },
+    requestOptions
+  );
+  // update the status keyword
+  feedback.typeKeywords = setEntityStatusKeyword(
+    feedback.typeKeywords,
+    feedback.status
+  );
+  feedback.typeKeywords = setDiscussableKeyword(
+    feedback.typeKeywords,
+    feedback.isDiscussable
+  );
+  feedback.typeKeywords = setDisplayMapKeyword(
+    feedback.typeKeywords,
+    feedback.displayMap
+  );
+  // get the backing item & data
+  const model = await getModel(feedback.id, requestOptions);
+  // create the PropertyMapper
+  const mapper = new PropertyMapper<Partial<IHubFeedback>, IModel>(
+    getPropertyMap()
+  );
+  // Note: Although we are fetching the model, and applying changes onto it,
+  // we are not attempting to handle "concurrent edit" conflict resolution
+  // but this is where we would apply that sort of logic
+  const modelToUpdate = mapper.entityToStore(feedback, model);
+  // update the backing item
+  const updatedModel = await updateModel(modelToUpdate, requestOptions);
+  // now map back into a feedback and return that
+  let updatedFeedback = mapper.storeToEntity(updatedModel, feedback);
+  updatedFeedback = computeProps(model, updatedFeedback, requestOptions);
+  // the casting is needed because modelToObject returns a `Partial<T>`
+  // where as this function returns a `T`
+  return updatedFeedback as IHubFeedback;
+}
+
+/**
+ * @private
+ * Remove a Hub Feedback object
+ * @param id
+ * @param requestOptions
+ */
+export async function deleteFeedback(
+  id: string,
+  requestOptions: IUserRequestOptions
+): Promise<void> {
+  const ro = { ...requestOptions, ...{ id } } as IUserItemOptions;
```

Comment:
I overlooked this until now, but there's a _lot_ of complexity related to successfully deleting a survey as compared to other entity types. This is because surveys are composed of up to 4 different items, which all live inside a folder in AGO, all of which need to be deleted (including the folder). Further complicating things, these items need to be deleted in a specific order, otherwise delete operations can fail. Additionally, there's the concept of "relationships" supported by the AGO platform which we lean on to identify/fetch the items. Also, items can be "protected" which means they can't be deleted without first issuing another XHR request to first remove that delete prevention. Given these complexities, we're likely going to need to port ~200 other lines of code to perform this seemingly simple task..

You can see what I mean by taking a look at the survey-service [removeItem](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-opendata-components/addon/services/survey-service.js#L741) method and the nightly E2E purge script's [deleteSurvey](https://github.com/ArcGIS/opendata-ui/blob/master/tests/hub-e2e/lib/helpers/survey.js#L40) method... 🤯 

To the best of my knowledge, we don't actually delete surveys from within the Hub UI anywhere, so I cannot speak to the reliability of the survey-service `removeItem` method, but I personally sunk a bunch of time into the nightly purge script's `deleteSurvey` to get that working as reliably as I could, though it looks like that is not removing delete prevention in it's current state (most likely b/c it's only targeting surveys created by E2E tests which wouldn't have delete prevention). We may want to consult with @dbouwman on whether we want/need to remove delete prevention here, maybe we don't need that piece given users need to explicitly add delete prevention from within the AGO UI...

Reply:
I verified we aren't currently handling the case in settings workspace panes where delete prevention is enabled for the entity item. The `delete` button is enabled, no call is made to remove delete prevention, and consequently the AGO API `delete` endpoint serves an error response, but the error is suppressed, and nothing happens in the UI. Options here include:

1. Disable `delete` button when entity's `protected` property is true + show a notice instructing the user to remove delete prevention
2. Call the endpoint to remove delete prevention before calling the `delete` endpoint

We may need direction from Dave and/or Sam on how to proceed. If the latter is preferred, we may need to create issues for the other entity types to update their delete logic.

We'll additionally need to consider how to approach for surveys given there are up to 4 items that could each have delete prevention enabled if the desire is the former.

Reply:
Resolving this convo. From the sounds of Brian's comment [here](https://devtopia.esri.com/dc/hub/issues/8849#issuecomment-4595995) it sounds like we're only on the hook for deleting the `form` item, not the feature service, either views, or the containing folder. Also it sounds like folks are already considering a pattern related to protected items, so we probably don't need to be concerned about that at this time either.